src/plugins/data/README.mdx

```diff
@@ -56,7 +56,73 @@ Coming soon.
 
 ## Index Patterns
 
-Coming soon.
+The Index Patterns API provides a consistent method of structuring and formatting documents
+and field lists across the various Kibana apps. Its typically used in conjunction with
+SearchSource for composing queries.
+
+### Index Patterns API
+
+- Get list of index patterns
+- Get default index pattern and examine fields
+- Get index pattern by id
+- Find index pattern by title
+- Create index pattern
+- Create index pattern and save it
+- Modify index pattern and save it
+- Delete index pattern
+
+#### Get list of index patterns
+
+```
+const idsAndTitles = await data.indexPatterns.getIdsWithTitle();
+idsAndTitles.forEach(({id, title}) => console.log(`Index pattern id: ${id} title: ${title}`));
+```
+
+#### Get default index pattern and examine fields
+
+```
+const defaultIndexPattern = await data.indexPatterns.getDefault();
+defaultIndexPattern.fields.forEach(({name}) => { console.log(name); })
+```
+
+#### Get index pattern by id
+
+```
+const id = 'xxxxxx-xxx-xxxxxx';
+const indexPattern = await data.indexPatterns.get(id);
+```
+
+#### Find index pattern by title
+
+```
+const title = 'kibana-*';
+const [indexPattern] = await data.indexPatterns.find(title);
+```
+
+#### Create index pattern
+
+```
+const indexPattern = await data.indexPatterns.create({ title: 'kibana-*' });
```

Comment:
Is there a way to save an index pattern after it was created?

Maybe we could add 

```
const indexPattern = await data.indexPatterns.create({ title: 'kibana-*' });
// Do something with the pattern
// Save it
```